bin/click-format

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+# ==============================================================================
+# click-format
+#
+#   Formatting helper script for click.
+#
+#   Takes a Hoon command and an optional list of Clay file dependencies (as
+#   paths or beams) and formats them as a text representation of a Hoon noun.
+#   When jammed, this noun can be passed to conn.c to execute the command on a
+#   running ship.
+#
+#   If a dependency file is passed as a beam, it will be interpreted as-is. If a
+#   dependency file is passed as a path, the default beak (i.e. [our base now])
+#   will be prepended to it.
+#
+#   The structure of a conn.c thread call looks like this:
+#
+#       [42 %fyrd [%base %hi %tape [%ship ~zod]]]
+#        1  2      3     4   5      6     7
+#
+#       1 = Unique ID for mapping results to requests
+#       2 = Command to conn.c routing the request
+#       3 = Desk in which thread lives
+#       4 = Name of thread
+#       5 = Mark for output type
+#       6 = Mark for input type
+#       7 = Input to thread as a noun
+#
+#   (1) - (6) are held constant by this client. (7) is provided by the user.
```

Comment:
I appreciate the thought put into composability here, but adding a second, largely stand-alone script complicates `bin/click` and also will make it more complicated to maintain `bin/click`. Is there a reason I'm overlooking that explains why a stand-alone script is better than a simple additional function in `bin/click`? Something like:
```bash
# Produce a text representation of a Hoon noun from a Hoon command and an
# optional list of Clay file dependencies (as paths or beams). The resulting
# noun can be jammed and then passed to conn.c to execute the command on a
# running ship.
#
# ...etc...
hoon_cmd_to_noun() {
  # Constants.
  REQ_ID=0
  CMD="%fyrd"
  DESK="%base"
  MARK_IN="%ted-eval"
  MARK_OUT="%noun"

  # Parse arguments.
  THREAD="$1" # %eval or %khan-eval
  DEPS_IDX=2
  DEPS_CNT=$(($# - $DEPS_IDX))
  if [[ $DEPS_CNT -eq 0 ]]; then
    TED_IN="[$MARK_IN '$THREAD']"
  else
    TED_IN="[$MARK_IN [$THREAD [${@:$DEPS_IDX:$DEPS_CNT} ~]]]"
  endif

  echo "[$REQ_ID $CMD [$DESK $THREAD $MARK_OUT $TED_IN]]"
}
```

Reply:
It wasn't a conscious decision, just an artifact of how I wrote click (first form requests correctly, then pipe them in correctly with various options).

That said, I think you're overstating the difficulty of supporting click because it's split into two scripts. Since the two scripts already both exist, I'd rather keep them as is, since I can imagine `click-format` being extended for `ovum` events. However, if you feel strongly about this then I'm also fine to merge them together, since the functionality of `bin/click-format` is minimal now and it's unclear what the future of click is, as compared to a fully-featured thick client.

Reply:
Has testing been easier because there are two scripts? My point is having `click-format` as a separate script implies it serves a purpose independent of `click`, whereas a function does not have that same implication, but if it's significantly easier to test when split in two, then that's a win. I agree supporting `click` is is trivial at this point, but I believe my point about the conceptual implications still stand. In either case, I don't think it's worth spending much more time on this question :)

Reply:
I would say that it was easier to test because it was split in two.

For now, I'll add a comment about possibly merging `click-format` into one script and link back to this discussion.